zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/application/AppConfig.kt

```diff
@@ -114,14 +116,33 @@ data class AppConfig(
      * If set [iconPath] will contain the path of the resource that points
      * to an icon image that will be used in the application window.
      */
-    val iconPath: String? = null
+    val iconPath: String? = null,
+    /**
+     * If set, contains custom properties that plugin authors can set and access.
+     */
+    internal val customProperties: Map<AppConfigKey<*>, Any> = emptyMap()
 ) {
 
     /**
      * Tells whether bounds check should be performed or not.
      * This depends on the various debug mode configurations.
      */
-    fun shouldCheckBounds() = debugMode.not() || (debugMode && debugConfig.relaxBoundsCheck.not())
+    fun shouldCheckBounds() = !debugMode || !debugConfig.relaxBoundsCheck
+
+    /**
+     * Retrieve a custom property set earlier using [AppConfigBuilder.withProperty]. If this property was
+     * never set, returns `null`.
+     *
+     * ### End Developers
+     *
+     * You probably don't need to call this API.
+     */
+    fun <T : Any> getProperty(key: AppConfigKey<T>): T? {
```

Comment:
I am tempted to say this should rather be a `operator fun get()` so you retrieve values via `config[myKey]`. But this might obscure the intention of this getter. Do you still know that you get custom properties with that? Maybe only when `AppConfigKey` had a different name like `CustomPropertyKey`...

Reply:
A `get` operator would definitely work here, but I'm trying to make it just a little harder to use to discourage people from using it :) except plugin authors who literally have to use it to get this functionality.

Reply:
I agree with @Baret on this. We should either use a getter or rename it to not resemble one. Maybe `findProperty`? Another thing is that we used `Maybe<T>` up until now instead of `T?`. I know that it is not a Kotlinesque solution but Java users might see `null`s here which are harder to handle than a `Maybe.empty()`. Whether we keep catering to Java users' needs is another discussion. We might decide at some point that this is ugly and Java users should learn Kotlin instead. :)

Reply:
Oh I forgot about `Maybe` ^^
I think this is the perfect use case. Maybe > nullable :)

Reply:
`T?` would be better from a Kotlin perspective but it's not good for Java users :(

Reply:
Based on the feedback here I've updated my version to this signature:
```kotlin
operator fun <T : Any> get(key: AppConfigKey<T>): Maybe<T>
```